cpp/src/toppra/geometric_path.cpp

```diff
@@ -10,4 +13,50 @@ Vectors GeometricPath::eval(const Vector &positions, int order) const {
     }
     return outputs;
   };
+
+
+Vector GeometricPath::proposeGridpoints(double max_err_threshold, int max_iteration, double max_seg_length, int min_nb_points) const {
+  std::forward_list<value_type> gridpoints {pathInterval()[0], pathInterval()[1]};
+  int nb_gridpoints = 2;
+
+  // Add points according to error threshold
+  for (auto iter=0; iter < max_iteration; iter++){
```

Comment:
Hum, rather strange use of `auto`...
```suggestion
  for (int iter=0; iter < max_iteration; iter++){
```